oasis-reference/src/main/kotlin/com/empowerops/volition/ref_oasis/ConnectionView.kt

```diff
@@ -0,0 +1,125 @@
+                }
+                hbox {
+                    label("Inputs: ")
+                    vbox {
+                        dataModel.simByName.getValue(it).inputs.forEach { input ->
+                            add(label("${input.name} [${input.lowerBound}, ${input.upperBound}] "))
+                        }
+                    }
+                }
+                hbox {
+                    label("Outputs: ")
+                    vbox {
+                        dataModel.simByName.getValue(it).outputs.forEach { output ->
+                            add(label(output.name))
+                        }
+                    }
+                }
+                hbox {
+                    spacing = 5.0
+                    button("refresh") {
+                        action {
+                            GlobalScope.launch {
+                                control.updateNode(it)
+                                dataModel.updateConfiguration(it)
+                            }
+                        }
+                    }
+                    button("delete") {
+                        action {
+                            GlobalScope.launch {
+                                dataModel.closeSim(it)
+                            }
+                        }
+                    }
+                    button("add setup") {
+                        action {
+                            GlobalScope.launch {
+                                dataModel.addConfiguration(it)
+                                dataModel.updateConfiguration(it)
+                            }
+                        }
+                    }
+                    button("remove setup") {
+                        action {
+                            GlobalScope.launch {
+                                dataModel.removeConfiguration(it)
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+
+    @Subscribe
+    fun whenNewNodeRegistered(event : PluginRegisteredEvent){
+        GlobalScope.launch(Dispatchers.JavaFx) {
+            regList.add(event.name)
+        }
+    }
+
+    @Subscribe
+    fun whenNodeUnRegistered(event : PluginUnRegisteredEvent){
+        GlobalScope.launch(Dispatchers.JavaFx) {
+            regList.remove(event.name)
+        }
+    }
+
+    @Subscribe
+    fun whenNodeUnRegistered(event : PluginRenamedEvent){
+        GlobalScope.launch(Dispatchers.JavaFx) {
+            regList.remove(event.oldName)
+            regList.add(event.newName)
+        }
+    }
+
+    @Subscribe
+    fun whenNodeRefreshed(event: PluginUpdatedEvent) {
+        GlobalScope.launch(Dispatchers.JavaFx) {
+            root.refresh()
+        }
+    }
+
+    fun bindDateModel(dataModel: DataModelService, control: OptimizerEndpoint, eventBus: EventBus) {
+        eventBus.register(this)
+        this.control = control
+        this.dataModel = dataModel
+        root.items = regList
```

Comment:
any reason not to do this in the constructor?

Reply:
I usually create the component in the initialization. The init flow is a bit different in the parent controller due to fxml loading. 
I think I can move these to constructor, move the creation to Main class and only pass the ListView to setData in OptimizerController.

Reply:
I did the change, now it is inited on the root and view is passed via the setData call after we have our data class ready, check Optimizer class for that change